viamonvifdiscovery/service_test.go

```diff
@@ -0,0 +1,92 @@
+// Package viamonvifdiscovery provides the discovery service that wraps ONVIF integration for the viamrtsp module
+package viamonvifdiscovery
+
+import (
+	"context"
+	"testing"
+
+	"github.com/viam-modules/viamrtsp"
+	"go.viam.com/rdk/logging"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/services/discovery"
+	"go.viam.com/test"
+)
+
+func TestDiscoveryService(t *testing.T) {
+	cfg := Config{}
+	ctx := context.Background()
+	logger := logging.NewTestLogger(t)
+
+	t.Run("Test Default Service", func(t *testing.T) {
+		testName := "test"
+		resourceCfg := resource.Config{API: discovery.API, Model: Model, Name: testName, ConvertedAttributes: &cfg}
+		dis, err := newDiscovery(ctx, nil, resourceCfg, logger)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, dis, test.ShouldNotBeNil)
+		test.That(t, dis.Name().ShortName(), test.ShouldResemble, testName)
+		test.That(t, dis.Close(ctx), test.ShouldBeNil)
+	})
+}
+
+func TestCredentials(t *testing.T) {
+	creds := []creds{
+		{Username: "", Password: ""},
+		{Username: "user1", Password: "pass1"},
+		{Username: "user2", Password: "pass2"},
+		{Username: "user1", Password: "pass3"},
+		{Username: "user1", Password: "pass4"},
+		{Username: "user2", Password: "pass5"},
+	}
+	dis := rtspDiscovery{Credentials: creds}
+	dis.countUniqueUsernames()
+	test.That(t, dis.Credentials[0].credNumber, test.ShouldEqual, 0)
+	test.That(t, dis.Credentials[1].credNumber, test.ShouldEqual, 0)
+	test.That(t, dis.Credentials[2].credNumber, test.ShouldEqual, 0)
+	test.That(t, dis.Credentials[3].credNumber, test.ShouldEqual, 1)
+	test.That(t, dis.Credentials[4].credNumber, test.ShouldEqual, 2)
+	test.That(t, dis.Credentials[5].credNumber, test.ShouldEqual, 1)
+	dis.Credentials[0].createName(0)
+	test.That(t, dis.Credentials[0].createName(6), test.ShouldEqual, "Camera_Insecure_6")
+	test.That(t, dis.Credentials[1].createName(7), test.ShouldEqual, "Camera_user1_7")
+	test.That(t, dis.Credentials[2].createName(8), test.ShouldEqual, "Camera_user2_8")
+	test.That(t, dis.Credentials[3].createName(9), test.ShouldEqual, "Camera_user1-1_9")
+	test.That(t, dis.Credentials[4].createName(10), test.ShouldEqual, "Camera_user1-2_10")
+	test.That(t, dis.Credentials[5].createName(11), test.ShouldEqual, "Camera_user2-1_11")
```

Comment:
[q] Walk me through the numbers at the end of the naming a little?

Reply:
If it’s an insecure camera, the camera name will be Camera_Insecure_[camera number]

If its a unique username/the first login credential that uses the username, the name will be Camera_[username]_[camera number]

If it’s a non unique username, the name will be Camera_[username]-[credNum]_[camera number]

this naming convention is what I randomly came up with, wnd if someone wants something else I’d be happy to change it. Only other thought I had was as to replace Camera with the Model that gets returned from the discovery service(or other info) but idk how much people will actually care